internal/tests/api/with-ns-registered/minipool_test.go

```diff
@@ -60,6 +60,85 @@ func TestMinipoolDepositAndStake(t *testing.T) {
 	defer nodeset_cleanup(snapshotName)
 
 	depositAndStakeMinipool(t)
+	simulateEthRewardToYieldDistributor(t)
+}
+
+// Utility function to send ETH and advance blockchain time
+func sendEthAndAdvanceTime(t *testing.T, address common.Address, slotsToAdvance int) {
+	sp := testMgr.GetConstellationServiceProvider()
+	txMgr := sp.GetTransactionManager()
+
+	sendEthOpts := &bind.TransactOpts{
+		From:  deployerOpts.From,
+		Value: big.NewInt(1e18),
+	}
+
+	sendEthTx := txMgr.CreateTransactionInfoRaw(address, nil, sendEthOpts)
+	testMgr.MineTx(t, sendEthTx, deployerOpts, "Sent ETH")
+
+	err := testMgr.AdvanceSlots(uint(slotsToAdvance), false)
+	require.NoError(t, err)
+	t.Logf("Advanced %d slots", slotsToAdvance)
+}
+
+// Simulate an ETH reward getting deposited to YieldDistributor
+func simulateEthRewardToYieldDistributor(t *testing.T) {
+	sp := testMgr.GetConstellationServiceProvider()
+	qMgr := sp.GetQueryManager()
+	slotsToAdvance := 1200 * 60 * 60 / 12
+
+	bindings, err := cstestutils.CreateBindings(testMgr.GetConstellationServiceProvider())
+	require.NoError(t, err)
+
+	// Get balances before harvest
+	var wethBalanceNodeBefore *big.Int
+	var wethBalanceTreasuryBefore *big.Int
+
+	err = qMgr.Query(func(mc *batch.MultiCaller) error {
+		bindings.Weth.BalanceOf(mc, &wethBalanceNodeBefore, nodeAddress)
+		bindings.Weth.BalanceOf(mc, &wethBalanceTreasuryBefore, bindings.TreasuryAddress)
+		return nil
+	}, nil)
+	require.NoError(t, err)
+
+	// Send 1 ETH to the deposit pool
+	sendEthAndAdvanceTime(t, bindings.DepositPoolAddress, slotsToAdvance)
+
+	// Send 1 ETH to the yield distributor
+	sendEthAndAdvanceTime(t, bindings.YieldDistributor.Address, 0)
+
+	// Call harvest()
+	harvestTx, err := bindings.YieldDistributor.Harvest(nodeAddress, big.NewInt(0), big.NewInt(1), deployerOpts)
+	require.NoError(t, err)
+	testMgr.MineTx(t, harvestTx, deployerOpts, "Called harvest from YieldDistributor")
+
+	// Again - to simulate an interval tick for rewards to go to treasury
+
+	// Send 1 ETH to the deposit pool
+	sendEthAndAdvanceTime(t, bindings.DepositPoolAddress, slotsToAdvance)
+
+	// Send 1 ETH to the yield distributor
+	sendEthAndAdvanceTime(t, bindings.YieldDistributor.Address, 0)
+
+	// Call harvest()
+	harvestTx, err = bindings.YieldDistributor.Harvest(nodeAddress, big.NewInt(1), big.NewInt(2), deployerOpts)
+	require.NoError(t, err)
+	testMgr.MineTx(t, harvestTx, deployerOpts, "Called harvest from YieldDistributor")
+
+	// Get balances after harvest
+	var wethBalanceNodeAfter *big.Int
+	var wethBalanceTreasuryAfter *big.Int
+
+	err = qMgr.Query(func(mc *batch.MultiCaller) error {
+		bindings.Weth.BalanceOf(mc, &wethBalanceNodeAfter, nodeAddress)
+		bindings.Weth.BalanceOf(mc, &wethBalanceTreasuryAfter, bindings.TreasuryAddress)
+		return nil
+	}, nil)
+	require.NoError(t, err)
+
+	// Verify balances
+	require.Equal(t, 1, wethBalanceNodeAfter.Cmp(wethBalanceNodeBefore))
```

Comment:
This is probably up to you and Mike but I suspect that having some kind of precalculated "expected rewards" to compare against would be more useful in terms of contract testing than just ensuring the balance increased.

Reply:
Yes, wander mentioned that he will include these in the docs during my last 1-1 with him. For now i'm just asserting greater than until he gives me exact numbers.